src/oop/programs_with_multiple_classes/hard/q2/MainTest.java

```diff
@@ -0,0 +1,191 @@
+       Person 2: Person{height=63.0}
+       Fort: Blanket{size='Large', threadCount=1500}
+     */
+
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral(personUc + " 1: " + personUc + "\\{" + height + "="),
+                new StringLiteral("75.0"),
+                new StringLiteral("\\}"),
+                new NewLine(),
+                new StringLiteral(personUc + " 2: " + personUc + "\\{" + height + "="),
+                new DoubleLiteral(),
+                new StringLiteral("\\}"),
+                new NewLine(),
+                new StringLiteral("Fort: " + blanketUc + "\\{" + size + "="),
+                new StringLiteral("'Large', "),
+                new StringLiteral(threadCount + "="),
+                new IntegerLiteral(1500),
+                new StringLiteral("\\}"),
+        };
+    }
+
+    public void runMain() {
+        TestFort.main(new String[0]);
+    }
+
+    /**
+     * Person Tests
+     **/
+    @Test
+    public void personClassHasCorrectFields() {
+        String missingFieldMessage = "Your " + personLc + " class is missing a required field.";
+        String incorrectFieldModifierMessage = "A field in your " + personLc + " class has an incorrect modifier.";
+        assertTrue(person.hasField(height, double.class), missingFieldMessage);
+        assertTrue(person.hasModifier(height, "private"), incorrectFieldModifierMessage);
+    }
+
+    @Test
+    public void personClassHasCorrectConstructors() {
+        String missingConstructorMessage = "Your " + personLc + " class does not have a required constructor.";
+        String incorrectConstructorModifierMessage =
+                "One of the constructors in your " + personLc + " class does not have the correct modifiers.";
+        Class<?>[] constructorClasses = new Class[]{
+                double.class
+        };
+        assertTrue(person.hasConstructor(constructorClasses), missingConstructorMessage);
+        assertTrue(person.hasModifier(constructorClasses, "public"), incorrectConstructorModifierMessage);
+    }
+
+    private static Stream<Double> personInputProvider() {
+        return Stream.of(
+                75.0,
+                63.0,
+                55.0,
+                80.0
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("personInputProvider")
+    public void personConstructorInitializesValuesCorrectly(Double height) throws Throwable {
+        Object[][] arguments = new Object[][]{
+                {height, double.class}
+        };
+        Object personInstance = person.createInstance(arguments);
+        _assertEquals(height, person.getFieldValue(personInstance, this.height),
+                "Your " + personLc + " constructor does not correctly initialize the " + this.height + " field.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("personInputProvider")
+    public void correctPersonToStringMethod(Double height) throws Throwable {
```

Comment:
Ideally, if we get the thing where we can prevent the students from editing certain portions of the code, we can remove this (but just keep it for now)

Reply:
I kept both in for now